### PR TITLE
add some notes on setting up dockerized databases for online testing

### PR DESCRIPTION
**Pull request by mprins**
This fixes and updates the "Online Tests" document hosted at https://docs.geotools.org/latest/developer/conventions/test/online.html

Adding some instructions on running dockerized versions of SQL Server and Oracle XE databases.

Note that this assumes #2890 is merged as far as the oracle part is concerned.

## Checklist

> Reviewing is a process done by project maintainers, mostly on a volunteer basis. We try to keep the overhead as small as possible and appreciate if you help us to do so by completing the following items. Feel free to ask in a comment if you have troubles with any of them.

For all pull requests:

- [x] Confirm you have read the [contribution guidelines](https://github.com/geotools/geotools/blob/master/CONTRIBUTING.md) 
- [x] You have sent a Contribution Licence Agreement (CLA) as necessary (not required for small changes, e.g., fixing typos in documentation)
- [x] Make sure the first PR targets the master branch, eventual backports will be managed later. This can be ignored if the PR is fixing an issue that only happens in a specific branch, but not in newer ones.
- [x] The changes are not breaking the build in downstream projects using SNAPSHOT dependencies, GeoWebCache and GeoServer.

The following are required only for core and extension modules (they are welcomed, but not required, for unsupported modules):
- [ ] There is an issue in [Jira](https://osgeo-org.atlassian.net/projects/GEOT) describing the bug/task/new feature (a notable exemptions is, changes not visible to end users). The ticket is for the GeoTools project, if the issue was found elsewhere it's a good practice to link to the origin ticket/issue.
- [x] The pull request contains changes related to a single objective. If multiple focuses cannot be avoided, each one is in its own commit and has a separate ticket describing it.
- [x] PR for bug fixes and small new features are presented as a single commit
- [ ] Commit message must be in the form "[GEOT-XYZW] Title of the Jira ticket"
- [ ] New unit tests have been added covering the changes
- [x] This PR passes all existing unit tests (test results will be reported by travis-ci after opening this PR)
- [x] This PR passes the [QA checks](https://docs.geotools.org/latest/developer/conventions/code/qa.html) (QA checks results will be reported by travis-ci after opening this PR)
- [x] Documentation has been updated accordingly.

Submitting the PR does not require you to check all items, but by the time it gets merged, they should be either satisfied or inapplicable.

### CODE REVIEW

File: docs/developer/conventions/test/online.rst [hunk 1]
```diff
@@ -95,20 +95,124 @@ Example fixture ~/.geotools/postgis/local.properties::
...
+
+    C:\\Documents and Settings\\Fred\>mkdir .geotools
+
+
+3. And set up any fixtures you need::
+
+    C:\\Documents and Settings\\Fred>cd .geotools
+    C:\\Documents and Settings\\Fred\\.geotools>mkdir postgis
+    C:\\Documents and Settings\\Fred\\.geotools>cd postgis
+    C:\\Documents and Settings\\Fred\\.geotools\\postgis>notepad typical.properties
```
Comment (jodygarnett) [inline docs/developer/conventions/test/online.rst:108]: Many test cases write out a template in `~/.geotools/` so you have something to edit.

File: docs/developer/conventions/test/online.rst [hunk 2]
```diff
@@ -95,20 +95,124 @@ Example fixture ~/.geotools/postgis/local.properties::
...
+
+You can use `docker <https://www.docker.com/>`_ to run a database flavour of your choice,
+some examples are given below. Also the Travis-CI build script shows how to install / run
+SQL Server and Oracle XE (see `.travis.yml <https://github.com/geotools/geotools/blob/master/.travis.yml>`_).
+Using docker will prevent the hassle of local installation on your computer possibly messing up your configuration.
+
+Oracle XE
+_________
+
+Oracle Database Express Edition (XE) is an unsupported version
```
Comment (jodygarnett) [inline docs/developer/conventions/test/online.rst:130]: These instructions may be good to have in the README.md of the oracle plugin itself.

File: docs/developer/conventions/test/online.rst (hunk 1 above)
Comment (mprins) [reply docs/developer/conventions/test/online.rst:108]: this is just a formatting fix in the existing doc as you can see in  https://docs.geotools.org/latest/developer/conventions/test/online.html it renders as shown below

image

File: docs/developer/conventions/test/online.rst (hunk 2 above)
Comment (mprins) [reply docs/developer/conventions/test/online.rst:130]: I disagree, the readme's are not exposed through search engines and having both will lead to one of them always being forgotten.

The readme's have little or no exposure and there is no consensus on what one should be able to find in the readme of a module - if one exists at all - eg. neither the jdbc-oracle nor jdbc-sqlserver even have a readme

in fact the only jdbc module with a readme is jdbc-hana and contains no such information.

Comment (aaime) [reply docs/developer/conventions/test/online.rst:130]: +1 on better search engine support

**Comment by aaime** [comment]
No further feedback from @jodygarnett , moving on and merging.

**Pull request merged**